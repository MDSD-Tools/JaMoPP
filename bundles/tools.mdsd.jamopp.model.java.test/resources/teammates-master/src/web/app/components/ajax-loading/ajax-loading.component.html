<div class="loading-container">
  <div class="spinner-border spinner-border-sm text-{{useBlueSpinner ? 'primary' : color}}" role="status"></div>
</div>
