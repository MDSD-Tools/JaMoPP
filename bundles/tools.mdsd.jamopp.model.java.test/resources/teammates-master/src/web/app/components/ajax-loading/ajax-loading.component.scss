.loading-container {
  display: inline;
  margin-right: 10px;
}
