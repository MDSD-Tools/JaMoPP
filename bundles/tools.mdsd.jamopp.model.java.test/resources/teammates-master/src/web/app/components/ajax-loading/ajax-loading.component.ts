import { Component, Input, OnInit } from '@angular/core';

/**
 * Displaying the ajax loader.
 */
@Component({
  selector: 'tm-ajax-loading',
  templateUrl: './ajax-loading.component.html',
  styleUrls: ['./ajax-loading.component.scss'],
})
export class AjaxLoadingComponent implements OnInit {

  @Input()
  useBlueSpinner: boolean = false;

  @Input()
  color: string = 'white';

  constructor() { }

  ngOnInit(): void {
  }

}
