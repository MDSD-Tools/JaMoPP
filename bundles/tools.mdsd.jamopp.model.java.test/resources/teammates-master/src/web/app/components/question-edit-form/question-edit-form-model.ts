import {
  FeedbackParticipantType,
  FeedbackQuestionDetails,
  FeedbackQuestionType,
  FeedbackVisibilityType,
  NumberOfEntitiesToGiveFeedbackToSetting,
} from '../../../types/api-output';

/**
 * The mode of operation for question edit form.
 */
export enum QuestionEditFormMode {
  /**
   * Adding a new feedback question.
   */
  ADD,

  /**
   * Editing the existing feedback question.
   */
  EDIT,
}

/**
 * The form model of question edit form.
 */
export interface QuestionEditFormModel {
  feedbackQuestionId: string;

  questionNumber: number;
  questionBrief: string;
  questionDescription: string;

  isQuestionHasResponses: boolean;

  questionType: FeedbackQuestionType;
  questionDetails: FeedbackQuestionDetails;

  giverType: FeedbackParticipantType;
  recipientType: FeedbackParticipantType;

  numberOfEntitiesToGiveFeedbackToSetting: NumberOfEntitiesToGiveFeedbackToSetting;
  customNumberOfEntitiesToGiveFeedbackTo: number;

  showResponsesTo: FeedbackVisibilityType[];
  showGiverNameTo: FeedbackVisibilityType[];
  showRecipientNameTo: FeedbackVisibilityType[];

  isUsingOtherFeedbackPath?: boolean;
  commonVisibilitySettingName?: string;
  isUsingOtherVisibilitySetting?: boolean;

  isDuplicating: boolean;
  isDeleting: boolean;
  isEditable: boolean;
  isSaving: boolean;
  isCollapsed: boolean;
  isVisibilityChanged: boolean;
  isFeedbackPathChanged: boolean;
  isQuestionDetailsChanged: boolean;
}
