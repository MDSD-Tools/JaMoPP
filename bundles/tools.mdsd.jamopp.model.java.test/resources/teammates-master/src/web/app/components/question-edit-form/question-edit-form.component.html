<div id="question-form-{{ model.questionNumber }}" class="card">
  <div class="card-header bg-primary text-white cursor-pointer" (click)="triggerModelChange('isCollapsed', !model.isCollapsed)">
    <div class="collapse-caret">
      <tm-panel-chevron [isExpanded]="!model.isCollapsed"></tm-panel-chevron>
    </div>
    <div id="question-header">
      <strong>Question <span id="question-number" *ngIf="!model.isEditable">{{ model.questionNumber }}</span></strong>
      <select id="question-number-dropdown" class="form-control question-number-select" [ngModel]="model.questionNumber" (ngModelChange)="triggerModelChange('questionNumber', $event)" *ngIf="model.isEditable" (click)="$event.stopPropagation()">
        <option *ngFor="let i of range(numOfQuestions)" [ngValue]="i + 1">{{ i + 1 }}</option>
      </select>
      {{ model.questionType | questionTypeName }}
      <ng-container *ngIf="model.isCollapsed">
        <br>
        <div class="collapse-qn-description">{{ model.questionBrief }}</div>
      </ng-container>
    </div>
    <div class="card-header-btn-toolbar">
      <div *ngIf="formMode === QuestionEditFormMode.EDIT">
        <button id="btn-edit-question" type="button" class="btn btn-primary btn-sm" *ngIf="!model.isEditable"
                (click)="triggerModelChangeBatch({'isEditable': true, 'isCollapsed': false}); $event.stopPropagation();"
                [disabled]="model.isSaving || model.isDeleting || model.isDuplicating">
          <i class="fas fa-pencil-alt"></i> Edit
        </button>
        <button id="btn-save-question" type="button" class="btn btn-primary btn-sm" *ngIf="model.isEditable"
                (click)="saveQuestionHandler(); $event.stopPropagation();"
                [disabled]="model.isSaving || model.isDeleting || model.isDuplicating">
          <tm-ajax-loading *ngIf="model.isSaving"></tm-ajax-loading><i class="fas fa-check"></i> Save
        </button>
        <button type="button" class="btn btn-primary btn-sm" *ngIf="model.isEditable"
                (click)="discardChangesHandler(false); $event.stopPropagation();"
                [disabled]="model.isSaving || model.isDeleting || model.isDuplicating">
          <i class="fas fa-ban"></i> Cancel
        </button>
        <button id="btn-delete-question" type="button" class="btn btn-primary btn-sm"
                (click)="deleteCurrentQuestionHandler(); $event.stopPropagation();"
                [disabled]="model.isSaving || model.isDeleting || model.isDuplicating">
          <tm-ajax-loading *ngIf="model.isDeleting"></tm-ajax-loading><i class="fas fa-trash"></i> Delete
        </button>
        <button id="btn-duplicate-question" type="button" class="btn btn-primary btn-sm" container="body"
                (click)="duplicateCurrentQuestionHandler(); $event.stopPropagation();"
                [disabled]="model.isSaving || model.isDeleting || model.isDuplicating"
                ngbTooltip="Make a copy of the existing question and add to the current feedback session."
                container="body">
          <tm-ajax-loading *ngIf="model.isDuplicating"></tm-ajax-loading><i class="far fa-copy"></i> Duplicate
        </button>
      </div>
      <div *ngIf="formMode === QuestionEditFormMode.ADD">
        <button type="button" class="btn btn-primary btn-sm"
                (click)="discardChangesHandler(true); $event.stopPropagation();"
                [disabled]="model.isSaving || model.isDeleting || model.isDuplicating">
          <i class="fas fa-ban"></i>
          Cancel
        </button>
      </div>
    </div>
  </div>
  <div *ngIf="!model.isCollapsed" @collapseAnim>
    <div class="card-body">
      <div class="background-color-light-blue">
        <div class="row">
          <div class="col-12">
            <div class="row padding-15px text-left">
              <div class="col-md-2">
                <div class="question-basic-info">Question</div>
              </div>
              <div class="col-md-10">
                <textarea id="question-brief" class="form-control" [ngModel]="model.questionBrief" (ngModelChange)="triggerModelChange('questionBrief', $event)" [disabled]="!model.isEditable"
                          placeholder="A concise version of the question e.g. &quot;How well did the team member communicate?&quot;"
                ></textarea>
              </div>
            </div>
            <div class="row padding-15px text-left">
              <div class="col-md-2">
                <div class="question-basic-info">[Optional]<br/>Description</div>
              </div>
              <div class="col-md-10">
                <tm-rich-text-editor [richText]="model.questionDescription" (richTextChange)="triggerModelChange('questionDescription', $event)" [isDisabled]="!model.isEditable" minHeightInPx="100"
                                     placeholderText="More details about the question e.g. &quot;In answering the question, do consider communications made informally within the team, and formal communications with the instructors and tutors.&quot;"
                ></tm-rich-text-editor>
              </div>
            </div>
            <div class="padding-15px">
              <tm-text-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.TEXT" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-text-question-edit-details-form>
              <tm-contribution-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.CONTRIB" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-contribution-question-edit-details-form>
              <tm-mcq-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.MCQ" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-mcq-question-edit-details-form>
              <tm-msq-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.MSQ" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-msq-question-edit-details-form>
              <tm-num-scale-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.NUMSCALE" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-num-scale-question-edit-details-form>
              <tm-rank-options-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.RANK_OPTIONS" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-rank-options-question-edit-details-form>
              <tm-rank-recipients-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.RANK_RECIPIENTS" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-rank-recipients-question-edit-details-form>
              <tm-rubric-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.RUBRIC" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-rubric-question-edit-details-form>
              <tm-constsum-options-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.CONSTSUM_OPTIONS" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-constsum-options-question-edit-details-form>
              <tm-constsum-recipients-question-edit-details-form *ngIf="model.questionType === FeedbackQuestionType.CONSTSUM_RECIPIENTS" [details]="model.questionDetails" (detailsChange)="triggerModelChange('questionDetails', $event)" [isEditable]="model.isEditable"></tm-constsum-recipients-question-edit-details-form>
            </div>
          </div>
        </div>
      </div>
      <tm-feedback-path-panel [model]="model" [commonFeedbackPaths]="commonFeedbackPaths"
                              [allowedFeedbackPaths]="allowedFeedbackPaths"
                              (customFeedbackPath)="triggerModelChange('isUsingOtherFeedbackPath', $event)"
                              (customNumberOfEntitiesToGiveFeedbackTo)="triggerModelChange('customNumberOfEntitiesToGiveFeedbackTo', $event)"
                              (numberOfEntitiesToGiveFeedbackToSetting)="triggerModelChange('numberOfEntitiesToGiveFeedbackToSetting', $event)"
                              (triggerModelChangeBatch)="triggerModelChangeBatch($event)"></tm-feedback-path-panel>
      <tm-visibility-panel [model]="model" [isCustomFeedbackVisibilitySettingAllowed]="isCustomFeedbackVisibilitySettingAllowed"
                           [commonFeedbackVisibilitySettings]="commonFeedbackVisibilitySettings"
                           (customVisibilitySetting)="triggerModelChange('isUsingOtherVisibilitySetting', $event)"
                           (triggerModelChangeBatch)="triggerModelChangeBatch($event)"
                           [visibilityStateMachine]="visibilityStateMachine"></tm-visibility-panel>
      <div class="row margin-top-15px" *ngIf="!isDisplayOnly">
        <div class="col-12 text-right">
          <button type="button" class="btn btn-light btn-margin-right" *ngIf="model.isEditable"
                  (click)="discardChangesHandler(false)"
                  [disabled]="model.isSaving || model.isDeleting || model.isDuplicating">
            <i class="fas fa-ban"></i> Cancel
          </button>
          <button class="btn btn-success"
              [disabled]="!model.isEditable || model.isSaving || model.isDeleting || model.isDuplicating"
              (click)="saveQuestionHandler()">
            <tm-ajax-loading *ngIf="model.isSaving"></tm-ajax-loading>
            <span *ngIf="formMode === QuestionEditFormMode.EDIT" ><i class="fas fa-check"></i> Save Changes</span>
            <span id="btn-save-new" *ngIf="formMode === QuestionEditFormMode.ADD"><i class="fas fa-check"></i> Save Question</span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
