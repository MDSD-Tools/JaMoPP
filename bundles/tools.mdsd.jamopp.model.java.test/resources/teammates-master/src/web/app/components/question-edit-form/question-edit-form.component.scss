.question-number-select {
  height: 32px;
  width: auto;
  display: inline;
  margin-left: 6px;
  margin-right: 5px;
}

.background-color-light-blue {
  background-color: #EAEFF5;
}

.padding-15px {
  padding: 15px;
}

.question-basic-info {
  padding-left: 15px;
  font-weight: bold;
}

.padding-left-15px {
  padding-left: 15px;
}

.margin-top-15px {
  margin-top: 15px;
}

.collapse-caret {
  margin-right: 10px;
}

.collapse-qn-description {
  font-weight: 300;
}
