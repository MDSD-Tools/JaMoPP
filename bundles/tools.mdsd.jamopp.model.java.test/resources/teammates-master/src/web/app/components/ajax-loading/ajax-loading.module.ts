import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AjaxLoadingComponent } from './ajax-loading.component';

/**
 * Ajax loading module.
 */
@NgModule({
  imports: [
    CommonModule,
  ],
  declarations: [
    AjaxLoadingComponent,
  ],
  exports: [
    AjaxLoadingComponent,
  ],
})
export class AjaxLoadingModule { }
