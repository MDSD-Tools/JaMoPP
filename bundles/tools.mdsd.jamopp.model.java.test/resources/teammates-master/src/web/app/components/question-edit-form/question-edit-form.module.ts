import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { NgbCollapseModule, NgbDropdownModule, NgbTooltipModule } from '@ng-bootstrap/ng-bootstrap';

import { AjaxLoadingModule } from '../ajax-loading/ajax-loading.module';
import { FeedbackPathPanelModule } from '../feedback-path-panel/feedback-path-panel.module';
import { PanelChevronModule } from '../panel-chevron/panel-chevron.module';
import {
  QuestionEditDetailsFormModule,
} from '../question-types/question-edit-details-form/question-edit-details-form.module';
import { RichTextEditorModule } from '../rich-text-editor/rich-text-editor.module';
import { TeammatesCommonModule } from '../teammates-common/teammates-common.module';
import { VisibilityMessagesModule } from '../visibility-messages/visibility-messages.module';
import { VisibilityPanelModule } from '../visibility-panel/visibility-panel.module';

import { QuestionEditFormComponent } from './question-edit-form.component';

/**
 * Module for all question edit UI in session edit page.
 */
@NgModule({
  imports: [
    CommonModule,
    FormsModule,
    NgbCollapseModule,
    NgbDropdownModule,
    NgbTooltipModule,
    AjaxLoadingModule,
    VisibilityMessagesModule,
    TeammatesCommonModule,
    RichTextEditorModule,
    QuestionEditDetailsFormModule,
    PanelChevronModule,
    FeedbackPathPanelModule,
    VisibilityPanelModule,
  ],
  declarations: [
    QuestionEditFormComponent,
  ],
  exports: [
    QuestionEditFormComponent,
  ],
})
export class QuestionEditFormModule { }
