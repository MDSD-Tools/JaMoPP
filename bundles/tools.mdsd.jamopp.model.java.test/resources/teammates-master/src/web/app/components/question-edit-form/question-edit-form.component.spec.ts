import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HttpClientTestingModule } from '@angular/common/http/testing';
import { FormsModule } from '@angular/forms';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { AjaxLoadingModule } from '../ajax-loading/ajax-loading.module';
import { FeedbackPathPanelModule } from '../feedback-path-panel/feedback-path-panel.module';
import {
  QuestionEditDetailsFormModule,
} from '../question-types/question-edit-details-form/question-edit-details-form.module';
import { RichTextEditorModule } from '../rich-text-editor/rich-text-editor.module';
import { TeammatesCommonModule } from '../teammates-common/teammates-common.module';
import { VisibilityMessagesModule } from '../visibility-messages/visibility-messages.module';
import { VisibilityPanelModule } from '../visibility-panel/visibility-panel.module';
import { QuestionEditFormComponent } from './question-edit-form.component';

import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { mockTinyMceUuid } from '../../../test-helpers/mock-tinymce-uuid';
import { PanelChevronModule } from '../panel-chevron/panel-chevron.module';

import { EXAMPLE_ESSAY_QUESTION_MODEL } from '../../pages-help/instructor-help-page/instructor-help-questions-section/instructor-help-questions-data';
import { QuestionEditFormModel } from './question-edit-form-model';

describe('QuestionEditFormComponent', () => {
  let component: QuestionEditFormComponent;
  let fixture: ComponentFixture<QuestionEditFormComponent>;

  mockTinyMceUuid();

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        QuestionEditFormComponent,
      ],
      imports: [
        HttpClientTestingModule,
        FormsModule,
        TeammatesCommonModule,
        AjaxLoadingModule,
        RichTextEditorModule,
        QuestionEditDetailsFormModule,
        NgbModule,
        FeedbackPathPanelModule,
        VisibilityMessagesModule,
        VisibilityPanelModule,
        BrowserAnimationsModule,
        PanelChevronModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(QuestionEditFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('set up with EXAMPLE_ESSAY_QUESTION_MODEL', () => {
    component.formModel = EXAMPLE_ESSAY_QUESTION_MODEL;
    const model: QuestionEditFormModel = component.model;
    expect(model).toBe(EXAMPLE_ESSAY_QUESTION_MODEL);
    expect(model.isUsingOtherFeedbackPath).toBeFalsy();
    expect(model.commonVisibilitySettingName).toBeTruthy();
    expect(model.isUsingOtherVisibilitySetting).toBeFalsy();
    expect(component.isCustomFeedbackVisibilitySettingAllowed).toBeTruthy();
  });

  it('triggerModelChange with EXAMPLE_ESSAY_QUESTION_MODEL', () => {
    component.formModel = EXAMPLE_ESSAY_QUESTION_MODEL;
    component.formModelChange.subscribe((data: QuestionEditFormModel) => {
      component.formModel = data;
    });
    const testStr: string = 'Hello World';
    component.triggerModelChange('questionNumber', 2);
    component.triggerModelChange('questionBrief', testStr);
    component.triggerModelChange('questionDescription', testStr);
    component.triggerModelChange('isQuestionHasResponses', true);

    const model: QuestionEditFormModel = component.model;
    expect(model.questionNumber).toBe(2);
    expect(model.questionBrief).toBe(testStr);
    expect(model.questionDescription).toBe(testStr);
    expect(model.isQuestionHasResponses).toBeTruthy();
  });

  it('should snap with default view', () => {
    expect(fixture).toMatchSnapshot();
  });
});
