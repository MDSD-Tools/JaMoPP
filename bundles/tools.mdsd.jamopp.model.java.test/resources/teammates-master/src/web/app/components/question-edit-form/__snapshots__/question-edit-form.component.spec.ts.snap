// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionEditFormComponent should snap with default view 1`] = `
<tm-question-edit-form
  FeedbackParticipantType={[Function Object]}
  FeedbackQuestionType={[Function Object]}
  FeedbackVisibilityType={[Function Object]}
  NumberOfEntitiesToGiveFeedbackToSetting={[Function Object]}
  QuestionEditFormMode={[Function Object]}
  VisibilityControl={[Function Object]}
  allowedFeedbackPaths={[Function Map]}
  commonFeedbackPaths={[Function Map]}
  commonFeedbackVisibilitySettings={[Function Array]}
  createNewQuestionEvent={[Function EventEmitter]}
  deleteCurrentQuestionEvent={[Function EventEmitter]}
  discardExistingQuestionChangesEvent={[Function EventEmitter]}
  discardNewQuestionEvent={[Function EventEmitter]}
  duplicateCurrentQuestionEvent={[Function EventEmitter]}
  feedbackQuestionsService={[Function FeedbackQuestionsService]}
  formMode={[Function Number]}
  formModelChange={[Function EventEmitter]}
  isDisplayOnly="false"
  model={[Function Object]}
  numOfQuestions="0"
  saveExistingQuestionEvent={[Function EventEmitter]}
  simpleModalService={[Function SimpleModalService]}
  visibilityStateMachine={[Function VisibilityStateMachine]}
>
  <div
    class="card"
    id="question-form-0"
  >
    <div
      class="card-header bg-primary text-white cursor-pointer"
    >
      <div
        class="collapse-caret"
      >
        <tm-panel-chevron>
          <div
            class="chevron"
          >
            
            
            <i
              class="fas fa-chevron-up"
            />
          </div>
        </tm-panel-chevron>
      </div>
      <div
        id="question-header"
      >
        <strong>
          Question 
          
          <span
            id="question-number"
          >
            0
          </span>
        </strong>
        
         Essay question 
        
      </div>
      <div
        class="card-header-btn-toolbar"
      >
        
        <div>
          
          <button
            class="btn btn-primary btn-sm"
            id="btn-edit-question"
            type="button"
          >
            <i
              class="fas fa-pencil-alt"
            />
             Edit 
          </button>
          
          
          <button
            class="btn btn-primary btn-sm"
            id="btn-delete-question"
            type="button"
          >
            
            <i
              class="fas fa-trash"
            />
             Delete 
          </button>
          <button
            class="btn btn-primary btn-sm"
            container="body"
            id="btn-duplicate-question"
            ngbtooltip="Make a copy of the existing question and add to the current feedback session."
            type="button"
          >
            
            <i
              class="far fa-copy"
            />
             Duplicate 
          </button>
        </div>
        
      </div>
    </div>
    
    <div
      class="ng-trigger ng-trigger-collapseAnim"
      style=""
    >
      <div
        class="card-body"
      >
        <div
          class="background-color-light-blue"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <div
                class="row padding-15px text-left"
              >
                <div
                  class="col-md-2"
                >
                  <div
                    class="question-basic-info"
                  >
                    Question
                  </div>
                </div>
                <div
                  class="col-md-10"
                >
                  <textarea
                    class="form-control ng-untouched ng-pristine ng-valid"
                    disabled=""
                    id="question-brief"
                    placeholder="A concise version of the question e.g. \\"How well did the team member communicate?\\""
                  />
                </div>
              </div>
              <div
                class="row padding-15px text-left"
              >
                <div
                  class="col-md-2"
                >
                  <div
                    class="question-basic-info"
                  >
                    [Optional]
                    <br />
                    Description
                  </div>
                </div>
                <div
                  class="col-md-10"
                >
                  <tm-rich-text-editor
                    minheightinpx="100"
                    placeholdertext="More details about the question e.g. \\"In answering the question, do consider communications made informally within the team, and formal communications with the instructors and tutors.\\""
                  >
                    <div
                      inviewport=""
                    >
                      
                    </div>
                  </tm-rich-text-editor>
                </div>
              </div>
              <div
                class="padding-15px"
              >
                
                <tm-text-question-edit-details-form
                  style=""
                >
                  <div
                    class="row padding-15px"
                  >
                    <div
                      class="col-md-10 allow-rich-text"
                    >
                      <label>
                        <input
                          disabled=""
                          type="checkbox"
                        />
                        <a>
                           Allow rich-text formatting (e.g., bold text, links, bullet lists, colors, etc.) in the answers
                        </a>
                      </label>
                    </div>
                    <div
                      class="col-12 question-recommended-length"
                    >
                       [Optional] 
                      <span
                        class="ngb-tooltip-class"
                        ngbtooltip="The recommended length is shown to the respondent but not enforced"
                      >
                         Recommended length 
                      </span>
                       for the response: 
                      <input
                        class="form-control ng-untouched ng-pristine ng-valid"
                        disabled=""
                        id="recommended-length"
                        min="1"
                        type="number"
                      />
                       words 
                    </div>
                  </div>
                </tm-text-question-edit-details-form>
                
                
                
                
                
                
                
                
                
              </div>
            </div>
          </div>
        </div>
        <tm-feedback-path-panel>
          <div
            class="background-color-light-green margin-top-15px padding-15px"
          >
            <div>
              <b
                class="feedback-path-title"
              >
                Feedback Path
              </b>
               (Who is giving feedback about whom?) 
            </div>
            <div
              autoclose="outside"
              class="margin-top-15px dropdown"
              ngbdropdown=""
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="btn btn-light white-space-normal dropdown-toggle"
                disabled=""
                id="btn-feedback-path"
                ngbdropdowntoggle=""
              >
                
                <span
                  style=""
                >
                   Students in this course will give feedback on 
                  <i
                    class="fas fa-arrow-right"
                  />
                   Other students in the course 
                </span>
                
              </button>
              <ul
                class="dropdown-menu"
                id="feedback-path-dropdown"
                ngbdropdownmenu=""
                x-placement="bottom"
              >
                <li
                  class="dropdown-header"
                >
                  Common feedback path combinations
                </li>
                
                <li
                  class="dropdown-divider"
                />
                <li
                  class="dropdown-item"
                >
                  Custom feedback path...
                </li>
              </ul>
            </div>
            
            
            <div
              class="row margin-top-15px align-items-center"
              style=""
            >
              <div
                class="offset-1 col-md-4 text-md-right"
              >
                 The maximum number of students each respondent should give feedback to: 
              </div>
              <div
                class="col-md-2"
              >
                <div
                  class="input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <div
                      class="input-group-text"
                    >
                      <input
                        class="ng-untouched ng-pristine ng-valid"
                        disabled=""
                        id="custom-recipients"
                        type="radio"
                      />
                    </div>
                  </div>
                  <input
                    class="form-control ng-untouched ng-pristine ng-valid"
                    disabled=""
                    id="custom-recipients-number"
                    min="1"
                    type="number"
                  />
                </div>
              </div>
              <div
                class="col-md-2"
              >
                <div
                  class="form-check"
                >
                  <label
                    class="form-check-label"
                  >
                    <input
                      class="form-check-input ng-untouched ng-pristine ng-valid"
                      disabled=""
                      id="unlimited-recipients"
                      type="radio"
                    />
                     Unlimited 
                  </label>
                </div>
              </div>
            </div>
          </div>
        </tm-feedback-path-panel>
        <tm-visibility-panel>
          <div
            class="margin-top-15px padding-15px background-color-light-green"
          >
            <div>
              
              <b
                class="visibility-title"
              >
                Visibility
              </b>
               (Who can see the responses?) 
            </div>
            <div
              class="margin-top-15px dropdown"
              ngbdropdown=""
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="btn btn-light white-space-normal dropdown-toggle"
                disabled=""
                id="btn-question-visibility"
                ngbdropdowntoggle=""
              >
                
                <span
                  style=""
                >
                    
                </span>
                
              </button>
              <ul
                class="dropdown-menu"
                id="question-visibility-dropdown"
                ngbdropdownmenu=""
                x-placement="bottom"
              >
                <li
                  class="dropdown-header"
                >
                  Common visibility options
                </li>
                
                
                <li
                  class="dropdown-divider"
                  style=""
                />
                
                <li
                  class="dropdown-item"
                  style=""
                >
                  Custom visibility options...
                </li>
              </ul>
            </div>
            
            <div
              class="margin-top-15px visibility-message"
            >
               This is the visibility hint as seen by the feedback giver: 
              <ul
                class="text-muted background-color-warning"
              >
                
                
                
                
                
                
                
                
                
                
                
                
                
                
                
                
                
                
                <li
                  style=""
                >
                  No-one can see your responses
                </li>
              </ul>
            </div>
          </div>
        </tm-visibility-panel>
        
        <div
          class="row margin-top-15px"
          style=""
        >
          <div
            class="col-12 text-right"
          >
            
            <button
              class="btn btn-success"
              disabled=""
            >
              
              
              <span>
                <i
                  class="fas fa-check"
                />
                 Save Changes
              </span>
              
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</tm-question-edit-form>
`;
